streams/src/main/java/org/apache/kafka/streams/errors/InvalidAdvanceIntervalException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.errors;
+/**
+ * Indicates wrong window advancement duration, which means the advancement is not in a range of zero to duration
+ *
+ * @see org.apache.kafka.streams.kstream.TimeWindows
+ */
+public class InvalidAdvanceIntervalException extends StreamsException {
```

Comment:
I think this is a public API change and would require a KIP. I also think, we don't need this, but can just extend the exception message of `IllegalArgumentException` to be more descriptive.

Reply:
removed the class in new commit.